### PR TITLE
refact: update with release branch

### PR DESCRIPTION
**Pull request by imbajin**
update from release, merge after #2061

### CODE REVIEW

File: hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeEdge.java [hunk 1]
```diff
@@ -369,7 +369,7 @@ public void vertices(HugeVertex owner, HugeVertex other) {
         if (ownerLabel.equals(this.label.sourceLabel())) {
             this.vertices(true, owner, other);
         } else {
-            // TODO: why when compare the label but ignore the result?
+            // TODO: why compare the label but ignore the result?
```
Comment (javeme) [inline hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeEdge.java:370]: seems missing assert

File: hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeVertex.java [hunk 2]
```diff
@@ -462,10 +464,14 @@ public <V> VertexProperty<V> property(
             E.checkArgument(!this.hasProperty(propertyKey.id()),
                             "Can't update primary key: '%s'", key);
         }
+        if (value == null) {
+            this.removeProperty(propertyKey.id());
+            return EmptyVertexProperty.instance();
+        }
```
Comment (javeme) [inline hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeVertex.java:468]: prefer to move this block into PR https://github.com/apache/incubator-hugegraph/pull/206 for consistent logic.

Comment (imbajin) [reply hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeVertex.java:468]: > prefer to move this block into PR #206 for consistent logic.

Not quite clear, something wrong?

image

File: hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeEdge.java [hunk 3]
```diff
@@ -362,6 +367,7 @@ public void vertices(HugeVertex owner, HugeVertex other) {
         if (ownerLabel.equals(this.label.sourceLabel())) {
             this.vertices(true, owner, other);
         } else {
+            // TODO: why compare the label but ignore the result?
```
Comment (javeme) [inline hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeEdge.java:370]: assert ownerLabel.equals(this.label.targetLabel());

File: hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeVertex.java (hunk 2 above)
Comment (javeme) [reply hugegraph-core/src/main/java/org/apache/hugegraph/structure/HugeVertex.java:468]: that's ok if it has been cherry-picked

**Comment by javeme** [comment]
we can add the 3 tests to filter file:
```java
Error:  Errors: 
Error:    HasTest$Traversals>HasTest.g_V_hasXp_neqXvXX:704 » NoIndex Don't accept query ...
Error:    PartitionStrategyProcessTest.shouldWriteToMultiplePartitions:465 » IllegalArgument
Error:    PartitionStrategyProcessTest.shouldWriteVerticesToMultiplePartitions:366 » IllegalArgument
[INFO] 
Error:  Tests run: 815, Failures: 0, Errors: 3, Skipped: 113
```

**Review by zyxxoo — APPROVED** [review]

**Review by javeme — APPROVED** [review]
tinkerpop hugegraph test 2023-01-17 00 56 50

tinkerpop hugegraph test timeout 2023-01-17 21 47 31

**Pull request merged**